test/blackbox-tests/test-cases/pkg/ocamlformat/dev-tool-deps-conflict-project-deps.t

```diff
@@ -0,0 +1,221 @@
+  $ mkpkg ocamlformat 0.26.2 <<EOF
+  > depends: [
+  >  "printer" {= "1.0"}
+  > ]
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/ocamlformat.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum ocamlformat.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a project that uses the fake OCamlFormat:
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo)
+  >  (depends (printer (= 2.0))))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let () = Printer.print ()
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name foo)
+  >  (libraries printer))
+  > EOF
+  $ cat > .ocamlformat <<EOF
+  > version = 0.26.2
+  > EOF
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path "dev-tools.locks/ocamlformat")
+  >  (repositories mock))
+  > (lock_dir
+  >  (repositories mock))
+  > (repository
+  >  (name mock)
+  >  (source "file://$(pwd)/mock-opam-repository"))
+  > EOF
+
+Lock the to trigger package management
+  $ dune pkg lock
+  Solution for dune.lock:
+  - printer.2.0
+
+It shows that the project uses printer.2.0
+  $ dune exec -- foo
+  Hello world!
+
+Format foo.ml, 'dune fmt' uses printer.1.0 instead. There is no conflict with different 
+versions of the same dependency.
+  $ DUNE_CONFIG__LOCK_DEV_TOOL=enabled dune fmt
+  Solution for dev-tools.locks/ocamlformat:
+  - ocamlformat.0.26.2
+  - printer.1.0
+  File "foo.ml", line 1, characters 0-0:
+  Error: Files _build/default/foo.ml and _build/default/.formatted/foo.ml
+  differ.
+  Promoting _build/default/.formatted/foo.ml to foo.ml.
+  [1]
+  $ cat foo.ml
+  formatted
+
+Revert "foo.ml"
+  $ cat > foo.ml <<EOF
+  > let () = Printer.print ()
+  > EOF
+
+Now "dune-project" does not depend on printer, but the executable "foo" depends on it.
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo))
+  > EOF
+
+Relock the project.
+  $ dune pkg lock
+  Solution for dune.lock:
+  (no dependencies to lock)
+
+The OCamlFormat binary from the dev-tool.
+  $ ./_build/_private/default/.dev-tool/ocamlformat/ocamlformat/target/bin/ocamlformat foo.ml
+  formatted
+
+There is no leak here. It is not taking the "printer" lib from dev-tools.
+  $ dune exec -- foo
+  File "dune", line 3, characters 12-19:
+  3 |  (libraries printer))
+                  ^^^^^^^
+  Error: Library "printer" not found.
+  -> required by _build/default/.foo.eobjs/byte/dune__exe__Foo.cmi
+  -> required by _build/default/.foo.eobjs/native/dune__exe__Foo.cmx
+  -> required by _build/default/foo.exe
+  -> required by _build/install/default/bin/foo
+  [1]
+
+Now the executable "foo" does not depend on printer, but "foo.ml" always uses Printer module from
+"printer" lib.
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name foo))
+  > EOF
+
+The OCamlFormat binary from the dev-tool.
```

Comment:
I don't think it's necessary to test that the ocamlformat binary works correctly after each step.

Reply:
Can you update the comment to explain this.

Reply:
I removed it and preparing to merge some other concern with this.